lib/beaker-pe/install/pe_utils.rb

```diff
@@ -890,7 +890,198 @@ def config_hosts_for_proxy_access hosts
+              if result.exit_code == 0
+                logger.info(result.stdout)
+                break
+              end
+            end
+          end
+        end
+
+
+        # removes the install_puppet_agent_pe_promoted_repo_on method
+
+        # Install shared repo of the puppet-agent on the given host(s).  Downloaded from
+        # location of the form PE_PROMOTED_BUILDS_URL/PE_VER/puppet-agent/AGENT_VERSION/repo
+        #
+        # @param [Host, Array<Host>, String, Symbol] hosts    One or more hosts to act upon,
+        #                            or a role (String or Symbol) that identifies one or more hosts.
+        # @param [Hash{Symbol=>String}] opts An options hash
+        # @option opts [String] :puppet_agent_version The version of puppet-agent to install, defaults to 'latest'
+        # @option opts [String] :pe_ver The version of PE (will also use host['pe_ver']), defaults to '4.0'
+        # @option opts [String] :copy_base_local Directory where puppet-agent artifact
+        #                       will be stored locally
+        #                       (default: 'tmp/repo_configs')
+        # @option opts [String] :copy_dir_external Directory where puppet-agent
+        #                       artifact will be pushed to on the external machine
+        #                       (default: '/root')
+        # @option opts [String] :puppet_collection Defaults to 'PC1'
+        # @option opts [String] :pe_promoted_builds_url Base URL to pull artifacts from
+        #
+        # @note on windows, the +:ruby_arch+ host parameter can determine in addition
+        #   to other settings whether the 32 or 64bit install is used
+        #
+        # @example
+        #   install_puppet_agent_pe_promoted_repo_on(host, { :puppet_agent_version => '1.1.0.227', :pe_ver => '4.0.0-rc1'})
+        #
+        # @return nil
+        def install_puppet_agent_pe_promoted_repo_on(hosts, opts)
+          opts[:puppet_agent_version] ||= 'latest'
+
+          block_on hosts do |host|
+            pe_ver = host[:pe_ver] || opts[:pe_ver] || '4.0.0-rc1'
+            opts = sanitize_opts(opts)
+            opts[:download_url] =
+              "#{opts[:pe_promoted_builds_url]}/puppet-agent/#{pe_ver}/#{opts[:puppet_agent_version]}/repos"
+            opts[:copy_base_local]    ||= File.join('tmp', 'repo_configs')
+            opts[:copy_dir_external]  ||= host.external_copy_base
+            opts[:puppet_collection] ||= puppet_collection_for(:puppet_agent, opts[:puppet_agent_version])
+            add_role(host, 'aio') # we are installing agent, so we want aio role
+            release_path = opts[:download_url]
+            variant, version, arch, codename = host['platform'].to_array
+            copy_dir_local = File.join(opts[:copy_base_local], variant)
+            onhost_copy_base = opts[:copy_dir_external]
+
+            release_path_end, release_file, download_file =
+            host.pe_puppet_agent_promoted_package_info(
```

Comment:
I think the methods we are missing are the `pe_puppet_agent_promoted_package_info` and `pe_puppet_agent_promoted_package_install` below which were removed from beaker::host::unix::pkg?

Reply:
I've been trying to find `beaker::host::unix::pkg` but my github code searches have not worked so far, do you know where I should look?

Reply:
@Tobi Lehman: So beaker::host::unix::pkg is just a namespace. The repository is beaker.  You should find everything under lib/beaker/host/

Reply:
Found the commit where they were removed: https://github.com/tlehman/beaker/commit/cdaedad9b216070e8fe27c307d1686f09ee5f257